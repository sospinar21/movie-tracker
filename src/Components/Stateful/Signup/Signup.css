.signup-container {
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column; }
  .signup-container h2 {
    color: white; }

.signup {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #353434;
  height: 400px;
  width: 600px;
  border-radius: 10px; }

.signup-input {
  height: 38px;
  width: 300px;
  margin: 20px 8px;
  border: solid 2px #26a566;
  border-radius: 5px;
  padding-left: 10px;
  font-size: 20px; }

.name {
  margin-left: 42px; }

.email {
  margin-left: 46px; }

.signup-button {
  margin: 20px 0;
  width: 413px;
  height: 38px;
  font-size: 18px;
  background-color: #26a566;
  color: white;
  border-radius: 5px;
  transition: 0.3s;
  box-shadow: 0 10px 18px #494444;
  border: none;
  outline: none; }

.signup-button:hover {
  cursor: pointer;
  box-shadow: 0 6px 18px #777; }

h2 {
  color: #353434; }
