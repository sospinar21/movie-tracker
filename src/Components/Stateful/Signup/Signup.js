import React, { Component } from 'react';
import {signUp} from './../../../Helpers/apiCalls'; 
import './Signup.css';

export class Signup extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      status: ''
    };
  }

  handleSignup = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    const email = this.state.email.toLowerCase();
    event.preventDefault();
    const message = await signUp(this.state.name, this.state.email, this.state.password);
    this.signUpAuthorization(message);
  }

  signUpAuthorization = (message) => {
    if (!message.status) {
      this.setState({
        name:'',
        email: '',
        password: '',
        status: message.data
      });
    } else {
      this.setState({
        name:'',
        email: '',
        password: '',
        status: 'Account created, please log in'
      });
    }
  }


  render() {
    return (
      <div className='signup-container'>
        <h2>Sign Up Here!</h2>
        <form onSubmit={this.handleSubmit}
          className='signup'>
          <label>Name
            <input className='signup-input name'type="text" name='name' value={this.state.name} onChange={this.handleSignup}/>
          </label>
          <label>Email
            <input className='signup-input email'type="email" name='email' value={this.state.email} onChange={this.handleSignup}/>
          </label>
          <label>Password
            <input className='signup-input' type="password" name='password' value={this.state.password} onChange={this.handleSignup}/>
          </label>
          <button className='signup-button'>Sign Up</button>
        </form>
        <h2>{this.state.status}</h2>
      </div>
    );
  }
}

export default Signup;