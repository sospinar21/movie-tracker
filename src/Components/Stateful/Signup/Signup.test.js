import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import React from 'react';
import { Signup, mapStateToProps, mapDispatchToProps } from './Signup';
import { Redirect } from 'react-router-dom';
jest.mock('../../../Helpers/apiCalls', () => {
  return jest.fn().mockImplementation(() => {
    return {
      signUp: jest.fn().mockImplementation(() => Promise.resolve([{data1: "data"}, {data2: "data2"}]))
    };
  });
});


describe('Signup', () => {

  let userSignUp;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      user: {name: 'Jack'}
    }
    userSignUp = shallow(<Signup {...mockProps} />)
  })
  

  describe('Component', () => {
    it('matches snapshot', () => {

      expect(userSignUp).toMatchSnapshot();
    })
  });

  describe('handleSignup', () => {

    let event;

    beforeEach(() => {
      event = {
        target: {
          name: "user",
          value: "Tony"
        },
        preventDefault: () => {}
      };
    });

    it('should set state based of the params when called', async () => {

      await userSignUp.instance().handleSignup(event);
      
      expect(userSignUp.state('user')).toEqual('Tony');
    });
  });
});
