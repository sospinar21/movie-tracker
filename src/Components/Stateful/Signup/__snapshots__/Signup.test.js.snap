// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup
    user={
        Object {
            "name": "Jack",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sign Up Here!
</h2>,
        <form
          className="signup"
          onSubmit={[Function]}
>
          <label>
                    Name
                    <input
                              className="signup-input name"
                              name="name"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </label>
          <label>
                    Email
                    <input
                              className="signup-input email"
                              name="email"
                              onChange={[Function]}
                              type="email"
                              value=""
                    />
          </label>
          <label>
                    Password
                    <input
                              className="signup-input"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
          </label>
          <button
                    className="signup-button"
          >
                    Sign Up
          </button>
</form>,
        <h2 />,
      ],
      "className": "signup-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up Here!",
        },
        "ref": null,
        "rendered": "Sign Up Here!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Name
              <input
                            className="signup-input name"
                            name="name"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
</label>,
            <label>
              Email
              <input
                            className="signup-input email"
                            name="email"
                            onChange={[Function]}
                            type="email"
                            value=""
              />
</label>,
            <label>
              Password
              <input
                            className="signup-input"
                            name="password"
                            onChange={[Function]}
                            type="password"
                            value=""
              />
</label>,
            <button
              className="signup-button"
>
              Sign Up
</button>,
          ],
          "className": "signup",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name",
                <input
                  className="signup-input name"
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input name",
                  "name": "name",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email",
                <input
                  className="signup-input email"
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input email",
                  "name": "email",
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password",
                <input
                  className="signup-input"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input",
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "signup-button",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sign Up Here!
</h2>,
          <form
            className="signup"
            onSubmit={[Function]}
>
            <label>
                        Name
                        <input
                                    className="signup-input name"
                                    name="name"
                                    onChange={[Function]}
                                    type="text"
                                    value=""
                        />
            </label>
            <label>
                        Email
                        <input
                                    className="signup-input email"
                                    name="email"
                                    onChange={[Function]}
                                    type="email"
                                    value=""
                        />
            </label>
            <label>
                        Password
                        <input
                                    className="signup-input"
                                    name="password"
                                    onChange={[Function]}
                                    type="password"
                                    value=""
                        />
            </label>
            <button
                        className="signup-button"
            >
                        Sign Up
            </button>
</form>,
          <h2 />,
        ],
        "className": "signup-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up Here!",
          },
          "ref": null,
          "rendered": "Sign Up Here!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Name
                <input
                                className="signup-input name"
                                name="name"
                                onChange={[Function]}
                                type="text"
                                value=""
                />
</label>,
              <label>
                Email
                <input
                                className="signup-input email"
                                name="email"
                                onChange={[Function]}
                                type="email"
                                value=""
                />
</label>,
              <label>
                Password
                <input
                                className="signup-input"
                                name="password"
                                onChange={[Function]}
                                type="password"
                                value=""
                />
</label>,
              <button
                className="signup-button"
>
                Sign Up
</button>,
            ],
            "className": "signup",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name",
                  <input
                    className="signup-input name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input name",
                    "name": "name",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email",
                  <input
                    className="signup-input email"
                    name="email"
                    onChange={[Function]}
                    type="email"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Email",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input email",
                    "name": "email",
                    "onChange": [Function],
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password",
                  <input
                    className="signup-input"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Password",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input",
                    "name": "password",
                    "onChange": [Function],
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "signup-button",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`signUp Component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup
    addFavorites={[Function]}
    signUp={[Function]}
    user={
        Object {
            "name": "Jack",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sign Up Here!
</h2>,
        <form
          className="signup"
          onSubmit={[Function]}
>
          <label>
                    Name
                    <input
                              className="signup-input name"
                              name="name"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </label>
          <label>
                    Email
                    <input
                              className="signup-input email"
                              name="email"
                              onChange={[Function]}
                              type="email"
                              value=""
                    />
          </label>
          <label>
                    Password
                    <input
                              className="signup-input"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
          </label>
          <button
                    className="signup-button"
          >
                    Sign Up
          </button>
</form>,
        <h2 />,
      ],
      "className": "signup-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up Here!",
        },
        "ref": null,
        "rendered": "Sign Up Here!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Name
              <input
                            className="signup-input name"
                            name="name"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
</label>,
            <label>
              Email
              <input
                            className="signup-input email"
                            name="email"
                            onChange={[Function]}
                            type="email"
                            value=""
              />
</label>,
            <label>
              Password
              <input
                            className="signup-input"
                            name="password"
                            onChange={[Function]}
                            type="password"
                            value=""
              />
</label>,
            <button
              className="signup-button"
>
              Sign Up
</button>,
          ],
          "className": "signup",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name",
                <input
                  className="signup-input name"
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input name",
                  "name": "name",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email",
                <input
                  className="signup-input email"
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input email",
                  "name": "email",
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password",
                <input
                  className="signup-input"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-input",
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "signup-button",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sign Up Here!
</h2>,
          <form
            className="signup"
            onSubmit={[Function]}
>
            <label>
                        Name
                        <input
                                    className="signup-input name"
                                    name="name"
                                    onChange={[Function]}
                                    type="text"
                                    value=""
                        />
            </label>
            <label>
                        Email
                        <input
                                    className="signup-input email"
                                    name="email"
                                    onChange={[Function]}
                                    type="email"
                                    value=""
                        />
            </label>
            <label>
                        Password
                        <input
                                    className="signup-input"
                                    name="password"
                                    onChange={[Function]}
                                    type="password"
                                    value=""
                        />
            </label>
            <button
                        className="signup-button"
            >
                        Sign Up
            </button>
</form>,
          <h2 />,
        ],
        "className": "signup-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up Here!",
          },
          "ref": null,
          "rendered": "Sign Up Here!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Name
                <input
                                className="signup-input name"
                                name="name"
                                onChange={[Function]}
                                type="text"
                                value=""
                />
</label>,
              <label>
                Email
                <input
                                className="signup-input email"
                                name="email"
                                onChange={[Function]}
                                type="email"
                                value=""
                />
</label>,
              <label>
                Password
                <input
                                className="signup-input"
                                name="password"
                                onChange={[Function]}
                                type="password"
                                value=""
                />
</label>,
              <button
                className="signup-button"
>
                Sign Up
</button>,
            ],
            "className": "signup",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name",
                  <input
                    className="signup-input name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input name",
                    "name": "name",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email",
                  <input
                    className="signup-input email"
                    name="email"
                    onChange={[Function]}
                    type="email"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Email",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input email",
                    "name": "email",
                    "onChange": [Function],
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password",
                  <input
                    className="signup-input"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Password",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-input",
                    "name": "password",
                    "onChange": [Function],
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "signup-button",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
